turbopack/crates/turbo-tasks-backend/src/backend/operation/aggregation_update.rs

```diff
@@ -204,26 +226,71 @@ impl AggregatedDataUpdate {
                 }
             }
         }
+        for (collectible, count) in collectibles_update {
+            let mut added = false;
+            let mut removed = false;
+            update!(
+                task,
+                AggregatedCollectible {
+                    collectible: *collectible
+                },
+                |old: Option<i32>| {
+                    let old = old.unwrap_or(0);
+                    let new = old + *count;
+                    if old <= 0 && new > 0 {
+                        added = true;
+                    } else if old > 0 && new <= 0 {
+                        removed = true;
+                    }
+                    (new != 0).then_some(new)
+                }
+            );
+            if added || removed {
+                let ty = collectible.collectible_type;
+                let dependent: SmallVec<[TaskId; 4]> = get_many!(task, CollectiblesDependent { collectible_type, task } if *collectible_type == ty => *task);
+                if !dependent.is_empty() {
+                    queue.push(AggregationUpdateJob::Invalidate {
+                        task_ids: dependent,
+                    })
+                }
+            }
+            if added {
+                result.collectibles_update.push((*collectible, 1));
+            } else if removed {
+                result.collectibles_update.push((*collectible, -1));
+            }
```

Comment:
Are these branches guaranteed to be mutually exclusive? An `debug_assert!` might be nice.

Reply:
That's removed in a follow-up PR anyway.